src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -762,4 +819,58 @@ private function isUninitializedValueError(\Error $e): bool
         return str_starts_with($e->getMessage(), 'Typed property')
             && str_ends_with($e->getMessage(), 'must not be accessed before initialization');
     }
+
+    private function getNestedAttributes(string $class = null): array
+    {
+        if (!$this->classMetadataFactory || null === $class || !$this->classMetadataFactory->hasMetadataFor($class)) {
+            return [];
+        }
+
+        $properties = [];
+        $serializedPaths = [];
+        $classMetadata = $this->classMetadataFactory->getMetadataFor($class);
+        foreach ($classMetadata->getAttributesMetadata() as $name => $metadata) {
+            if (!$serializedPath = $metadata->getSerializedPath()) {
+                continue;
+            }
+            $pathIdentifier = implode(',', $serializedPath);
+            if (isset($serializedPaths[$pathIdentifier])) {
+                throw new LogicException(sprintf('Duplicate serialized path: "%s". Used for properties "%s" and "%s".', $pathIdentifier, $serializedPaths[$pathIdentifier], $name));
+            }
+            $serializedPaths[$pathIdentifier] = $name;
+            $properties[$name] = $serializedPath;
+        }
+
+        return $properties;
+    }
+
+    private function extractNestedValue($path, $data)
+    {
+        foreach ($path as $element) {
+            if (!isset($data[$element])) {
+                return null;
+            }
+            $data = $data[$element];
+        }
+
+        return $data;
+    }
+
+    private function setNestedValue($data, $elements, $value)
+    {
+        $element = array_shift($elements);
+        if (isset($data[$element]) && (!\is_array($data[$element]) || (\is_array($data[$element]) && !$elements))) {
+            throw new LogicException(sprintf('The element you are trying to set is already populated: "%s".', $element));
+        }
+        if (!$elements) {
+            $data[$element] = $value;
+        } else {
+            if (!isset($data[$element])) {
```

Comment:
instead of this "if", we can use `$data[$element] ?? []` on the call to setNestedValue